src/_P145_MQxxx.ino

```diff
@@ -0,0 +1,282 @@
+//
+// Conversion algorithm:
+// Sensor resistance is logaritmic to the concentration of a gas. Sensors are not specific to a single
+// gas. Key is the ratio between the measured Rs and a reference resistor Rzero (Rs/Rzero).
+// For each specific gas a somewhat linear relation between gas concentration and the ratio (Rs/Rzero) is
+// show on a log-log chart. Various algorithms are described in above mentioned literature. 
+// This plugin supports 3, almost similar, algorithms to make it easier to copy a parameter set from the
+// internet for a specific sensor/gas combination.
+// 
+// Calibration algorithm:
+// Each measurement calculate Rcal assuming the concentration is at the reference level
+// Remember the lowest value of Rcal assuming it belongs to measuring the lowest concentration 
+///################################### Configuration data ###############################################
+// This plugin uses the following predefined static data storage 
+// analog input reading which is now always A0.
+// PCONFIG_FLOAT_RLOAD  PCONFIG_FLOAT(0)  RLOAD   [Ohm] 
+// PCONFIG_FLOAT_RZERO  PCONFIG_FLOAT(1)  RZERO   [Ohm]
+// PCONFIG_FLOAT_REF    PCONFIG_FLOAT(2)  REF. level [ppm]
+// PCONFIG_FLAGS        PCONFIG(0)        Enable compensation & Enable Calibration
+// PCONFIG_TEMP_TASK    PCONFIG(1)        Temperature compensation task
+// PCONFIG_TEMP_VAL     PCONFIG(2)        Temperature compensation value
+// PCONFIG_HUM_TASK     PCONFIG(3)        Humidity compensation task
+// PCONFIG_HUM_VAL      PCONFIG(4)        Humidity compensation value
+// PCONFIG_SENSORT      PCONFIG(5)        Sensor type
+//#######################################################################################################
+
+#include "_Plugin_Helper.h"
+#ifdef USES_P145
+#include "src/PluginStructs/P145_data_struct.h"  // MQ-xxx sensor specific data
+
+#define PLUGIN_145
+#define PLUGIN_ID_145     145           // plugin id
+#define PLUGIN_NAME_145   "Gases - MQxxx (MQ135 CO2, MQ3 Alcohol) [TESTING]" // "Plugin Name" is what will be dislpayed in the selection list
+#define PLUGIN_VALUENAME1_145 "level"   // variable output of the plugin. The label is in quotation marks
+#define PLUGIN_145_DEBUG  false         // set to true for extra log info in the debug
+
+// Static, per plugin instance, data stored in the generic ESPeasy structures see usage defined above
+// PCONFIG(n)    : stores an integer (8)
+// PCONFIG_FLOAT : stors a float (4)
+// PCONFIG_LONG  : stores a long (4, shared with PCONFIG_ULONG)
+// PCONFIG_ULONG : stores an unsigned long (4, shared with PCONFIG_LONG)
+#define PCONFIG_FLOAT_RLOAD PCONFIG_FLOAT(0)
+#define PCONFIG_FLOAT_RZERO PCONFIG_FLOAT(1)
+#define PCONFIG_FLOAT_REF   PCONFIG_FLOAT(2)
+#define PCONFIG_FLAGS       PCONFIG(0)
+#define PCONFIG_TEMP_TASK   PCONFIG(1)
+#define PCONFIG_TEMP_VAL    PCONFIG(2)
+#define PCONFIG_HUM_TASK    PCONFIG(3)
+#define PCONFIG_HUM_VAL     PCONFIG(4)
+#define PCONFIG_SENSORT     PCONFIG(5)
```

Comment:
Please prefix these defines with the plugin NR.
These are rather generic names and when these would be used elsewhere in the code you would get all kinds of freaky issues.